### ISSUE TITLE
[Feature] Add project_id and/or project_ids to `asana__task`

### ISSUE THREAD

**Issue by irvingpop** [fivetran/dbt_asana#24]
### Is there an existing feature request for this?

- [X] I have searched the existing issues

### Describe the Feature

As a user of this model, I would've expected project_id to be an available field in the `asana__task` Model.  

I know this can be a little bit more complicated because a task could be associated to multiple projects, but still leaving it out entirely means you have to then join against a staging table (stg_asana__project_task) which is not ideal.

I'd be willing to contribute a PR for this if you're interested. 

### Are you interested in contributing this feature?

- [X] Yes.
- [ ] Yes, but I will need assistance and will schedule time during your [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours) for guidance.
- [ ] No.

**Comment by fivetran-sheringuyen** [comment]
Hey @irvingpop! This sounds like a feature that would be helpful to include in this package. I noticed that you are interested in contributing to this feature, if you wanted to give it a stab, we definitely encourage it! Our team can then review your work during a future sprint and add in whatever else is needed on our end required for release. In the meantime, please let us know if you have any questions during your PR process. How does that sound to you? :smile:

**Comment by irvingpop** [comment]
hey @fivetran-sheringuyen apologies for the long delay.  Here's the PR: https://github.com/fivetran/dbt_asana/pull/25

**Comment by fivetran-catfritz** [comment]
Hi @irvingpop to update you, we are working on this issue in the next couple of weeks. We'll keep you posted and let you know if we have any questions!

**Comment by fivetran-catfritz** [comment]
Hi @irvingpop I have created a test branch you can try it out! I made some small logic changes since we have to support several warehouses, and there was an issue with how it worked postgres. You can install the test branch using the below in place or your normal package installation:
```yml
packages:
- git: https://github.com/fivetran/dbt_asana.git
  revision: feature/add-project-id
  warn-unpinned: false
```
It would be very helpful and appreciated if you are able to try this out and let us know if you have any feedback!

**Comment by irvingpop** [comment]
hi @fivetran-catfritz thank you so much for picking this up! 

when modifying my packages.yml I noticed that `feature/add-project-id` didn't work for the revision, but `customer/add-project-id` did (matches the branch name in your PR) - do I have that right?

I'm doing a `dbt run` now and will let you know

**Comment by irvingpop** [comment]
results of dbt run look pretty good!  The only minor thing I'm struggling with is that the usage of `project_ids_array` is a bit awkward and took me a few permutations to get to this:

```sql
select split(project_ids, ', ') as proj_ids_array from analytics.dbt_irving_asana.ASANA__TASK
where array_contains('1202690727474195'::variant, proj_ids_array)
```

should we document this, or do you have any suggestions for using it less awkwardly?

**Comment by irvingpop** [comment]
> @irvingpop Thanks for sorting out the branch name and for the feedback! As for the split(), did that accomplish what you needed or was there something else you were hoping to do?

hey @fivetran-catfritz it does what I need, I mostly just curious if there were better ways to do this or if we should document this for future travelers.

**Comment by fivetran-catfritz** [comment]
Hi @irvingpop another one to try might be something like `unnest(string_to_array(project_ids, ', '))`. As for the documentation, thanks for the suggestion! I made sure to add to our docs that the delimiter is a comma + space for those lists. 

Also, I closed this issue since I released the code changes today, but if you have any other questions or suggestions feel free to continue this thread!
